rasa/importers/importer.py

```diff
@@ -251,10 +252,35 @@ async def get_domain(self) -> Domain:
         domains = [importer.get_domain() for importer in self._importers]
         domains = await asyncio.gather(*domains)
 
-        return reduce(
+        # Check if NLU data has any retrieval intents,
+        # if yes add corresponding retrieval actions with `utter_` prefix automatically to an empty domain.
+        nlu_data = await self.get_nlu_data()
+        if nlu_data.retrieval_intents:
+            domains.append(
+                self._get_domain_with_retrieval_actions(nlu_data.retrieval_intents)
+            )
+
+        combined_domain = reduce(
             lambda merged, other: merged.merge(other), domains, Domain.empty()
         )
 
+        # Make the domain known which intents are retrieval intents
+        combined_domain._update_retrieval_intent_properties(nlu_data.retrieval_intents)
```

Comment:
this should happen within the domain (e.g. `__init__`): we can't rely on the creator of the domain to call this method and forgetting to call it would lead to an inconsistent state.

Reply:
Refactored to be taken care of during domain merge.